src/main/java/org/apache/commons/compress/utils/ByteUtils.java

```diff
@@ -266,4 +270,17 @@ private static void checkReadLength(final int length) {
             throw new IllegalArgumentException("Can't read more than eight bytes into a long value");
         }
     }
+
+    public static byte[] utf16Decode(final char[] chars) {
+        if (chars == null) {
+            return null;
+        }
+        final ByteBuffer encoded = UTF_16LE.encode(CharBuffer.wrap(chars));
```

Comment:
Hm, why little-endian?

Reply:
Hi @garydgregory,

I don't have the answer, but the existing implementation of decryption use this 🤷